### ISSUE TITLE
How to convert_model -m "./mycustomodel"

### ISSUE THREAD

**Issue by christianjoun** [ELS-RD/transformer-deploy#54]
I have a custom model

`
class BERTClass(torch.nn.Module):
    def __init__(self, num_labels=4):
        super(BERTClass, self).__init__()
        self.l1 = BertModel.from_pretrained('bert-base-multilingual-uncased')
        self.l2 = torch.mean
        self.l3 = torch.nn.Linear(768, num_labels)

    
    def forward(self, ids, masks, token_type_ids):        
        last_hidden_state, _ = self.l1(ids, attention_mask=masks, token_type_ids=token_type_ids)
        avg_pooling = self.l2(last_hidden_state, dim=1)
        output = self.l3(avg_pooling)
        
        return output`

How do I go about saving my custom model so that I can run convert_model -m "./mycustomodel"?

Currently I am saving the model this way

`
    model_2_save = model.module if hasattr(model, "module") else model

    checkpoint = {
        'epoch': args.epochs,
        'num_labels': args.num_labels,
        'max_text_length': MAX_TEXT_LENGTH,
        'state_dict': model_2_save.state_dict()
    }

    torch.save(checkpoint, args.model_dir + "/pt_model.pt")`

Is it better to save the pretrained bit and convert separately from the fully connected layer and then combine them after conversion or do I need to drive my custom class from `PreTrainedModel` so as to be able to use `save_pretrained`?  Do you happen to have an example I can follow? Thanks for the amazing repo.

**Comment by pommedeterresautee** [comment]
I would perform the conversion outside the command line. 
There are some assumptions in the way the command line works which won't work in your case.
What you really need is to convert it to ONNX then to TensorRT (if you want TRT optimized version).

Then I would use a random supported model and convert it with the command line to generate all configurations, etc. for Triton and just replace the model with the one you really want to use.

Is it a custom model to generate embeddings? If yes, as a random model, take a sentence transformer one (like in the README).

**Comment by christianjoun** [comment]
Thank you
